lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,459 @@
+                iris.Constraint(time=this_time))
+            first_guess_time = iris_time_to_datetime(
+                first_guess_lightning_cube.coord('time').copy())[
+                    first_guess_lightning_cube.coord(
+                        'time'
+                        ).nearest_neighbour_index(this_point)]
+            first_guess_slice = first_guess_lightning_cube.extract(
+                iris.Constraint(time=first_guess_time))
+            err_string = "No matching {} cube for {}"
+            if not isinstance(prob_lightning_slice,
+                              iris.cube.Cube):
+                raise ConstraintMismatchError(
+                    err_string.format("lightning", this_time))
+            if abs((first_guess_time - this_time).total_seconds()) > 7201.:
+                raise ConstraintMismatchError(
+                    err_string.format("first-guess", this_time))
+            cube_slice.data = first_guess_slice.data
+            cube_slice.coord('forecast_period').convert_units('minutes')
+            fcmins = cube_slice.coord('forecast_period').points[0]
+
+            # Increase prob(lightning) to Risk 2 (pl_dict[2]) when within
+            #   lightning halo (lrt_lev2; 50km of an observed ATDNet strike):
+            cube_slice.data = np.where(
+                (prob_lightning_slice.data >= self.lrt_lev2) &
+                (cube_slice.data < self.pl_dict[2]),
+                self.pl_dict[2], cube_slice.data)
+            lratethresh = self.lrt_lev1(fcmins)
+
+            # Increase prob(lightning) to Risk 1 (pl_dict[1]) when within
+            #   lightning storm (lrt_lev1; ~5km of an observed ATDNet strike):
+            cube_slice.data = np.where(
+                (prob_lightning_slice.data >= lratethresh) &
+                (cube_slice.data < self.pl_dict[1]),
+                self.pl_dict[1], cube_slice.data)
+
+            new_cube_list.append(cube_slice)
+
+        new_prob_lightning_cube = new_cube_list.merge_cube()
+        new_prob_lightning_cube = check_cube_coordinates(
+            cube, new_prob_lightning_cube)
+
+        # Apply precipitation adjustments.
+        new_prob_lightning_cube = self.apply_precip(new_prob_lightning_cube,
+                                                    prob_precip_cube)
+
+        # If we have VII data, increase prob(lightning) accordingly.
+        if prob_vii_cube:
+            new_prob_lightning_cube = self.apply_ice(new_prob_lightning_cube,
+                                                     prob_vii_cube)
+        return new_prob_lightning_cube
+
+    def apply_precip(self, prob_lightning_cube, prob_precip_cube):
+        """
+        Modify Nowcast of lightning probability with precipitation rate
+        probabilities at thresholds of 0.5, 7 and 35 mm/h.
+
+        Args:
+            prob_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability.
+                This is modified in-place.
```

Comment:
Looks like it's only the forecast period coordinate unit that's modified - the function returns a new cube from a cubelist.  Could you update please.